packages/x-date-pickers/src/internals/hooks/usePicker/hooks/useValueAndOpenStates.ts

```diff
@@ -145,7 +145,7 @@ export function useValueAndOpenStates<
       ...prevState,
       // We reset the shallow value whenever we fire onChange.
       clockShallowValue: shouldFireOnChange ? undefined : prevState.clockShallowValue,
-      lastCommittedValue: shouldFireOnAccept ? value : prevState.lastCommittedValue,
+      lastCommittedValue: shouldFireOnAccept ? newValue : prevState.lastCommittedValue,
```

Comment:
It's strange that we didn't catch this before... 🤷 
We always had one step too "old" `lastCommittedValue` when controlling the value. 🙈